### ISSUE TITLE
mute menu

### ISSUE THREAD

**Issue by hoodiek** [glitch-soc/mastodon#119]
https://github.com/glitch-soc/mastodon/issues/118 related to linked issue, the main thing is a distinction between 'hard' and 'soft' mutes
hard mutes should hide notifications from person, and all notifications where they are in yr mentions and someone replies to them, also being in yr mentions

soft mutes should only ever remove the muted person from home timelines, alongside muting people @ ing that person.

also, if viewing a thread of something someone else posted, if muted person replies, soft mutes should be able to in context view view the thing person replied to said post from another.

a third checkbox should be 'mute posts this person boosts'

in summary, the three options should be as such when muting someone:
-mute from timelines (public and home)
-mute from notifications
-mute this person's boosts of other people

any combination should be allowed of these options

**Comment by dustine** [comment]
Separating what a mute targets seems like a really good idea: avoids having to define different future kinds of mutes (soft mute, hard mute, boost mute?) and instead make mutes customizable on a per-case basis! And probably also simplifies how to show what kind of mute someone has (filter per mute option, show ticklist of mute parameters), or changing it on the fly too.

Maybe this logic could be brought back into the feature itself, beyond just user-fronting?

**Comment by marrus-sh** [comment]
Are there still aspects of this which haven't been implemented? @beatrix-bitrot?
